### PR TITLE
Add/32676 category and tag tracks within products screen

### LINKED ISSUE #32676: Add categories and tags: new Tracks events
**Description:** we want to include Tracks events in the Categories, Tags and Attributes pages to learn more about how users interact with the features. 

**Acceptance criteria:**

- [ ] wcadmin_product_attributes_add
Description: fired when the user adds a new attribute
Event props:
page: attributes | product (identify if the attribute was created in the attributes page or within the add/edit product)
enable_archive: Yes|No
default_sort_order

- [ ] wcadmin_product_attributes_edit
Description: fired when the user adds a new attribute
Event props:
page: attributes | product (identify if the attribute was created in the attributes page or within the add/edit product)
enable_archive: Yes|No
default_sort_order

- [ ] wcadmin_product_attributes_configure_terms
Description: fired when the user clicks the “configure terms” option in the attributes list

- [ ] wcadmin_product_attributes_add_term
Description: fired when the user adds a new term

- [ ] wcadmin_product_tags_add
Description: fired when the user adds a new tags
Event props:
page: tags | product (identify if the tag was created in the tags page or within the add/edit product)

- [ ] wcadmin_product_category_add
Description: fired when the user adds a new category
Event props:
page: category | product (identify if the attribute was created in the category page or within the add/edit product)
parent_category: None|Uncategorized|Other
display_type: default|products|...
image: Yes|No

- [ ] wcadmin_product_category_manage
Description: fired when the user clicks on one of the manage options
Event props:
option_selected: edit| quick edit| delete...

- [ ] wcadmin_product_category_update
Description: fired when the user updates one of the categories

### PR DESCRIPTION
**Pull request by louwie17**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add tracks on the product editing and creation page for tags and attributes.
This is currently branched off of #33120, please look at the commits I made in the diff ([link](https://github.com/woocommerce/woocommerce/pull/33121/files/e946306585eafe9c9f0ce400a55299482f0ce54a..327c4f6d564bdf02649c022c696e4a18d1962ac0))

Closes #32676  .

### How to test the changes in this Pull Request:

1. Go to the product creation or edit page
2. And have tracks output enabled in your console: `localStorage.setItem( 'debug', 'wc-admin:*' );`
3. Click on the **Attributes** tab and select an attribute from the dropdown and click `Add`
4. Try adding a new term if you selected a none custom attribute, this should trigger the `wcadmin_product_attributes_add_term` track with the page param set to `product`
5. Click **Save attributes**, this should trigger a `wcadmin_product_attributes_add` with the page param set to `product`.

Testing product tags
1. Add a new tag on the right side (**product tags**), this should output the `wcadmin_product_tags_add` track.
2. Now click on **Choose from the most used tags** and add some of those tags, this should also trigger the `wcadmin_product_tags_add` track. Check if the extra params make sense for this.

Testing product categories
1. Add a new category on the right side (**product categories**), this should output the `wcadmin_product_category_add` track. This is triggered on the back-end, so you did have to have the [WooCommerce Admin Test Helper](https://github.com/woocommerce/woocommerce-admin-test-helper) enabled and look at the track logs under **WooCommerce > Status > Logs** for that day.

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33121/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Update use of wp_localize_script` acd46c6ac9790ec0d442fa7fb44ded9acfeb0519
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+			const tagInput = document.querySelector< HTMLInputElement >(
+				'#new-tag-product_tag'
+			);
+			if ( tagInput && tagInput.value && tagInput.value.length > 0 ) {
+				recordEvent( 'product_tags_add', {
+					page: 'product',
+					tag_string_length: tagInput.value.length,
+					tag_list_size:
+						( document.querySelector( '.tagchecklist' )?.children
+							.length || 0 ) + 1,
```
Comment (octaedro) [inline plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts:238]: I'm not sure about the + 1. After testing this on my end, `tag_list_size` tracks one more item than it should.

**Comment by octaedro** [comment]
For the event `wcadmin_product_category_add` we are not sending the props:

- description: fired when the user adds a new category
- parent_category: None|Uncategorized|Other
- display_type: default|products|...
- image: Yes|No

I think that it's probably expected, but I wanted to confirm.

**Comment by octaedro** [comment]
Should we also track the tag deletion?

Screen Capture on 2022-05-25 at 12-15-42

**Review by octaedro — COMMENTED** [review]
You did a great job here @louwie17. The code looks mainly good and it's testing well here.
Just left a couple of questions.

**Comment by louwie17** [comment]
> For the event `wcadmin_product_category_add` we are not sending the props:
> 
> * description: fired when the user adds a new category
> * parent_category: None|Uncategorized|Other
> * display_type: default|products|...
> * image: Yes|No
> 
> I think that it's probably expected, but I wanted to confirm.

Aaah yeah, good call out, I guess I only made those changes as part of #33118, I will copy those same changes to here. Also `description` is not a prop, that just describes the track within the issue.

**Comment by louwie17** [comment]
> Should we also track the tag deletion?

Yeah good suggestion.

File: plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts (hunk 1 above)
Comment (louwie17) [reply plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts:238]: Fixed as part of this commit: https://github.com/woocommerce/woocommerce/pull/33121/commits/8f61bef2e0e8b947dcc3b6cfccb49c160647ffdc

**Comment by louwie17** [comment]
@octaedro thanks for the review and suggestions, I updated this PR to account for them, if you have time to re-review this? :)

**Review by octaedro — DISMISSED** [review]
Thank you @louwie17 for addressing the changes. This is testing well here and code LGTM :shipit:

Just a small thing that looks unrelated to the changes added in this PR, I'm seeing this PHP Notice in my debug.log
```
[26-May-2022 17:47:57 UTC] PHP Notice:  Function WP_Scripts::localize was called <strong>incorrectly</strong>. The <code>$l10n</code> parameter must be an array. To pass arbitrary data to scripts, use the <code>wp_add_inline_script()</code> function instead. Please see <a href="https://wordpress.org/support/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 5.7.0.) in /wp-includes/functions.php on line 5831
```

**Comment by louwie17** [comment]
> Thank you @louwie17 for addressing the changes. This is testing well here and code LGTM :shipit:
> 
> Just a small thing that looks unrelated to the changes added in this PR, I'm seeing this PHP Notice in my debug.log
> 
> ```
> [26-May-2022 17:47:57 UTC] PHP Notice:  Function WP_Scripts::localize was called <strong>incorrectly</strong>. The <code>$l10n</code> parameter must be an array. To pass arbitrary data to scripts, use the <code>wp_add_inline_script()</code> function instead. Please see <a href="https://wordpress.org/support/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 5.7.0.) in /wp-includes/functions.php on line 5831
> ```

@octaedro good catch, this was not related to my changes in particular, but by the changes my PR depended on in https://github.com/woocommerce/woocommerce/pull/33120, if you wanted to re-test it? I fixed it as part of this commit: https://github.com/woocommerce/woocommerce/pull/33121/commits/acd46c6ac9790ec0d442fa7fb44ded9acfeb0519

**Review by octaedro — APPROVED** [review]
Thank you @louwie17! LGTM 🚀

**Pull request merged**